app/validators/customer_email_validator.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class CustomerEmailValidator < ActiveModel::Validator
+  DOMINIOS_PROIBIDOS = ['@gmail.com', '@yahoo.com.br', '@hotmail.com.br', '@iugu-lite.com.br'].freeze
+
+  def validate(record)
+    DOMINIOS_PROIBIDOS.each do |dominio|
+      if record.email.end_with?(dominio)
+        record.errors.add(:email, 'Domínio não permitido.')
+        break
+      end
+    end
```

Comment:
O próprio end_with? aceita receber mais de uma opção para testar, como são poucas opções a perfomance não ficaria afetada e você poderia simplificar com algo como:

```suggestion
      return unless record.email.end_with?(*BLOCKED_DOMAINS)
        
      record.errors.add(:email, 'Domínio não permitido.')
    end
```

O `*` é o [splat operator](https://www.campuscode.com.br/conteudos/single-splat-operator)

Reply:
Implementamos a guard clause, Henrique. Obrigado.